airbyte-integrations/bases/source-acceptance-test/source_acceptance_test/tests/test_core.py

```diff
@@ -150,6 +150,84 @@ def test_has_secret(self):
     def test_secret_never_in_the_output(self):
         """This test should be injected into any docker command it needs to know current config and spec"""
 
+    def test_secret_is_properly_marked(self, actual_connector_spec: ConnectorSpecification, detailed_logger):
+        def is_property_name_secret(path: str) -> Tuple[str, bool]:
```

Comment:
I thought of this option, decided to implement functions within functions because they are only used in one place and no need to expose them further. But I agree that for readability and testing it's better to declare them at class level

Reply:
Ok right! I'm not totally against this pattern, but in this case, I'd like these functions to be un-nested for testing